alf/optimizers/optimizers.py

```diff
@@ -192,6 +202,77 @@ def __init__(self,
             self.name = NewClsName + "_" + str(NewCls.counter)
             NewCls.counter += 1
 
+    @common.add_method(NewCls)
+    def _prepare_capacity_states(self, capacity_ratio: float):
+        """prepare capacity related states
+
+        Args:
+            capacity_ratio: the capacity ration specifying the ratio of learnable parameters
+                to the number of all parameters
+        Returns:
+            states: the states containing:
+                - old_param: the value of the parameters before gradient step if
+                    ``self._masked_out_value`` is not None
+        """
+        states = {}
+        if capacity_ratio < 1:
+            for param_group in self.param_groups:
+                for p in param_group['params']:
+                    s = {}
+                    if self._masked_out_value is None:
+                        # only save previous param value if masked_out_value is unspecified
+                        s['old_param'] = p.data.clone()
+                    states[p] = s
+        return states
```

Comment:
Given the current implementation, it's better to change this to `_clone_params()` and return a dict where the value is the cloned parameter.

Reply:
Changed function name and other relevant places accordlingly.